docs/src/reference/configuration.rst

```diff
@@ -0,0 +1,1033 @@
+Configuration parameters for Nextstrain SARS-CoV-2 workflow
+===========================================================
+
+S3_DST_BUCKET
+-------------
+
+-  type: string
+-  description: S3 bucket to store files from the ``upload`` rule in ``export_for_nextstrain.smk``. Currently only available to Nextstrain builds.
+
+S3_DST_COMPRESSION
+------------------
+
+-  type: string
+-  description: Compression format to use for files uploaded to S3 by the ``upload`` rule.
+-  examples
+
+   -  ``xz``
+   -  ``gz``
+
+S3_DST_ORIGINS
+--------------
+
+-  type: array
+-  items:
+
+   -  type: string
+
+-  description: List of input names (i.e., “origins”) for which intermediate files should be uploaded to S3 by the ``upload`` rule.
+-  examples
+
+   -  ``["gisaid"]``
+
+active_builds
+-------------
+
+-  type: string
+-  description: Comma-delimited list of names of builds to run (allowing a subset of all builds to be specified).
+-  examples
+
+   -  ``global``
+   -  ``global,africa,north-america``
+
+ancestral
+---------
+
+-  type: object
+-  description: Configuration of augur ancestral command that infers ancestral sequences based on a tree.
+
+inference
+~~~~~~~~~
+
+-  type: string
+-  description: Calculate joint or marginal maximum likelihood ancestral sequence states
+-  examples
+
+   -  ``joint``
+   -  ``marginal``
+
+builds
+------
+
+-  type: object
+-  description: Named builds to produce by the workflow from the given inputs. Builds are indexed by name and include any number of build attributes that can be used to control subsampling, Auspice configuration, and more.
+
+.. warning::
+
+   Build names currently only allow alpha characters, underscores, and hyphens (``A-Z``, ``a-z``, ``_``, ``-``), but must not contain ``tip-frequencies`` as it is a special string used for Nextstrain builds.
```

Comment:
I also noticed that just `tip-frequencies` passes the regex, is this fine? Or should it be matching for `_tip-frequencies`?

Reply:
We should err on the side of caution here and make sure the build name doesn't end with [any of the 5 suffixes we filter out in the nextstrain server](https://github.com/nextstrain/nextstrain.org/blob/d3a1fb7185ca3348c4121535484145db4619b714/src/utils/index.js#L83-L88): 

```js

  // All JSON files which aren't a sidecar file with a known suffix are assumed to
  // be v2+ JSONs (aka "unified" JSONs)
  const sidecarSuffixes = ["meta", "tree", "root-sequence", "seq", "tip-frequencies", "measurements"];
  const datasets = jsonFiles
    .filter((filename) => !sidecarSuffixes.some((suffix) => filename.endsWith(`_${suffix}.json`)))
```

Reply:
Added this to issue #900